tests/foreman/api/test_http_proxy.py

```diff
@@ -263,3 +263,49 @@ def test_positive_assign_http_proxy_to_products():
         assert r.http_proxy_id == http_proxy_b.id
 
     product_a.sync({'async': True})
+
+
+@pytest.mark.tier2
+def test_positive_sync_proxy_with_certificate(target_sat):
+    """Assign http_proxy with ca.crt to Repo and test
+     whether http_proxy and cert are used during sync.
+
+       Steps:
+       - Generate cert files with custom_cert_generate,
+         collect cacert path.
+       - Create new org, product, and http-proxy with cacert.
+       - Create repo using selected http proxy, perform repo sync.
+
+    :expectedresults: http-proxy with cacert is assigned to repo,
+        sync operation uses assigned http-proxy with the cert.
+
+    """
+    proxy_host = settings.http_proxy.auth_proxy_url.replace("http://", "").replace(":3128", "")
+
+    # Cleanup any existing certs that may conflict
+    target_sat.custom_certs_cleanup()
+    target_sat.custom_cert_generate(proxy_host)
+
+    # Create org, HTTP_Proxy, and repo
+    org = target_sat.api.Organization().create()
+    http_proxy = target_sat.api.HTTPProxy(
+        name=gen_string('alpha', 15),
+        url=settings.http_proxy.auth_proxy_url,
+        username=settings.http_proxy.username,
+        password=settings.http_proxy.password,
+        organization=[org],
+        cacert="/root/cacert.crt",
+    ).create()
+    product = target_sat.api.Product(organization=org).create()
+    repo = target_sat.api.Repository(
+        product=product,
+        http_proxy_policy='use_selected_http_proxy',
+        http_proxy_id=http_proxy.id,
+    ).create()
+    product.update()
+
+    assert repo.http_proxy_policy == 'use_selected_http_proxy'
+    assert repo.http_proxy_id == http_proxy.id
+
+    repo.sync()
```

Comment:
Probably want to add another assertion after this line to verify the sync worked.